internal/pkg/composable/controller.go

```diff
@@ -256,7 +281,8 @@ func (c *contextProviderState) Set(mapping map[string]interface{}) error {
 		return nil
 	}
 	c.mapping = mapping
-	c.signal <- true
+
+	go func() { c.signal <- true }()
```

Comment:
yes both can be true, but multiple writes mostly, this is triggered on event OnAdd

Reply:
What is the point of moving this into a go function? I would prefer not to create a go function on every change that will allocate a 4k stack for every little event with this change.

What are other options that we could do in this path? What is the real reason for this change?

Would the following work without creating a go routine:

```
select {
case c.signal <- true:
default:
}
```

Reply:
moving this to go function will make sure signal is not lost somewhere
when you have a controller handling other stuff like a config writing to signal will fail and signal will be lost.
this way signal is not lost and goroutines are quickly destroyed when draining a channel

Reply:
reason for this is that when notify is not read right away this will be hanging. with this hanging lock i not released and even `OnAdd` event receiver will be stuck on acquiring a lock

Reply:
If `c.signal` has a buffer size of 1 then we can ensure that it have a true sitting on the channel, which is all we need. So having the following:

```
select {
case c.signal <- true:
default:
}
```

Will ensure that at least one true is on the channel, and if another is already there then it will fallthrough to the default and continue on. This removes the need to have a goroutine.